WordPress/gutenberg
# Add publish flow in site editor

## What?
Part of: https://github.com/WordPress/gutenberg/issues/59878
Related: https://github.com/WordPress/gutenberg/issues/59878

This PR continues the work on unifying the publishing flow between post and site editors and adds to the site editor the `publish/save` button that is used in post editor. This means we have the same publishing flow between the two editors. Additionally the updated `Post Status` component is reused too.

Because of some extra use cases in site editor and trying to keep smaller change sets, this first iteration conditionally renders the reused `save/publish` button and panel and the existing one in site editor.

The extra use cases in site editor include:
1. Activating/previewing a theme (change of labels, callbacks, etc..)
2. The save button in site editor's sidebar (view mode) renders the save panel in a Modal
3. Some specific logic for how to handle the save when making changes to global styles page

Eventually all these should be consolidated by probably passing some extra props down to the reusable components or having an internal API to render something differently and will be explored separately. Also after unifying the flow we should update the designs based on: https://github.com/WordPress/gutenberg/issues/59878#issuecomment-2051395013



## Testing Instructions
1. Everything related to save and publish flow in post editor should work/look the same way as before.
2. In site editor if we are viewing pages test the save and publish flow with draft pages, etc.., multiple entities' changes
3. In site editor in view mode, the save button in the sidebar should work exactly as before, as it still renders the existing save button/panel.
4. Activating/previewing theme should work exactly as before as it still renders the existing components.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/16275880/c27919c2-6000-479b-b8f5-40cc05ef6c92
